modules/middleware/fetchPackage.js

```diff
@@ -5,15 +5,16 @@ const createPackageURL = require('../utils/createPackageURL');
 const createSearch = require('../utils/createSearch');
 const getNpmPackageInfo = require('../utils/getNpmPackageInfo');
 const incrementCounter = require('../utils/incrementCounter');
+const createRequestOptions = require('../utils/createRequestOptions');
 
 function tagRedirect(req, res) {
   const version = req.packageInfo['dist-tags'][req.packageVersion];
 
   res
-    .set({
+    .set(createRequestOptions({
       'Cache-Control': 'public, s-maxage=60, max-age=3600', // 1 minute at the CDN, 1 hour on clients
       'Cache-Tag': 'redirect, tag-redirect'
-    })
+    }, req.query.npmrc))
```

Comment:
Hey ! . really Liked the PR , 
Can you just deconstruct the `req.query` object like this
```bash
const {npmrc} = req.query ; 
```
It wont do much but it will help